webapp/src/app/shared/phaser/entities/cc-entity.ts

```diff
@@ -1,139 +1,179 @@
+	details: { level: { level: number, offset: number }, type: string, settings: any } = <any>{};
 	entitySettings: {
 		collType: string;
 		baseSize: Point3;
 		sheets: {
 			fix: {
-				gfx: string | Phaser.BitmapData;
+				gfx: any;
 				x: number;
 				y: number;
 				w: number;
 				h: number;
+				scaleX?: number;
+				scaleY?: number;
 				renderHeight?: number;
 				offsetX?: number;
 				offsetY?: number;
 				flipX?: boolean;
 				flipY?: boolean;
+				tint?: number;
+				alpha?: number;
 			}[],
 			offset?: Point;
 			renderMode?: string;
-			singleColor?: boolean;
-			flipX: boolean;
+			flipX?: boolean;
 		}
 		scalableX: boolean;
 		scalableY: boolean;
 		scalableStep: number;
 		pivot: Point;
 	} = <any>{};
 	
-	protected constructor(game: Phaser.Game, map: CCMap, x: number, y: number, inputEvents: InputEvents, typeName: string) {
-		super(game, 0, 0, null);
-		this.setInputEvents(inputEvents);
+	protected constructor(scene: Phaser.Scene, map: CCMap, x: number, y: number, typeName: string) {
+		super(scene, typeName, false);
+		scene.add.existing(this);
 		this.map = map;
-		game.add.existing(this);
+		this.container.x = Math.round(x);
+		this.container.y = Math.round(y);
 		this.details = <any>{
 			type: typeName
 		};
+	}
+	
+	
+	protected init(): void {
+		this.container = <any>this.scene.add.container(0, 0);
 		
-		this.boundingBoxOffsetGroup = game.add.group();
-		this.boundingBoxOffsetGroup.add(this);
-		
-		this.levelOffsetGroup = game.add.group();
-		this.levelOffsetGroup.add(this.boundingBoxOffsetGroup);
-		
-		this.group = game.add.group();
-		this.group.add(this.levelOffsetGroup);
-		
-		// actual coordinates of the entity
-		this.group.x = Math.round(x);
-		this.group.y = Math.round(y);
-		
-		const collImg = this.game.add.image();
+		const collImg = this.scene.add.graphics();
+		this.container.add(collImg);
 		this.collisionImage = collImg;
 		
 		collImg.alpha = 0;
-		this.levelOffsetGroup.add(collImg);
-		collImg.inputEnabled = false;
 		
-		// handle hover input
-		collImg.events.onInputOver.add(() => {
-			if (!this.selected) {
-				collImg.alpha = 0.35;
-			}
-		});
-		collImg.events.onInputOut.add(() => {
-			if (!this.selected) {
-				collImg.alpha = 0;
-			}
-		});
+		this.inputZone = this.scene.add.zone(0, 0, 50, 50);
+		this.inputZone.setOrigin(0);
+		this.inputZone.setData('entity', this);
+		this.container.add(this.inputZone);
 		
-		this.visible = false;
-		this.setEvents();
+		
+		this.inputZone.on('pointerover', () => this.inputOver());
+		this.inputZone.on('pointerout', () => this.inputOut());
+	}
+	
+	
+	public inputOver() {
+		if (!this.selected) {
+			this.collisionImage.alpha = 0.35;
+		}
+	}
+	
+	public inputOut() {
+		if (!this.selected) {
+			this.collisionImage.alpha = 0;
+		}
+	}
+	
+	protected activate(): void {
+		this.inputZone.setInteractive();
+	}
+	
+	
+	protected deactivate(): void {
+		this.inputZone.disableInteractive();
+	}
+	
+	
+	preUpdate(time: number, delta: number): void {
```

Comment:
why should they be prefixed? `_` is usually used for private variables

Reply:
Usually unused parameters are prefixed with `_` but I guess that in this case it's not recommended because the `preUpdate` should be the same everywhere.